services/terrain/src/terrain/core.clj

```diff
@@ -59,6 +60,12 @@
         (tc/with-logging-context {:user-info (cheshire/encode user-info)}
                                  (handler request))))))
 
+(defn- wrap-tree-urls-config
+  [handler]
+  (fn [request]
+    (with-tree-urls-base (config/tree-urls-base)
+      (handler request))))
```

Comment:
At some point in the future, we'd probably want to put a middleware namespace in the tree-urls-client package (or some related package). The middleware function wouldn't have to change much:

```